### ISSUE TITLE
Add  %GDSEARCH%

### ISSUE THREAD

**Issue by ceed0** [goldendict/goldendict#1645]
From https://github.com/xiaoyifang/goldendict-ng/pull/417

I can copy that pull request here, but I don't know Qt enough to tell if it's good.

**Comment by nonwill** [comment]
xiaoyifang/goldendict-ng#417 is not a good extensible implementation.
Maybe the `Compound expressions` feature which has been around there for a long time is exactly what you need:
![handict_web_file111](https://github.com/goldendict/goldendict/assets/46510529/55088f50-ef4c-453e-a44a-3b10b1b151f3)
Doing so does not need to make a big fuss to satisfy someones' desire. Aut of course, it is also possible to do these jobs in a Program dictionary by caching search history.

**Comment by nonwill** [comment]
xiaoyifang/goldendict-ng#417 is not a good extensible implementation.
Maybe the `Compound expressions` feature which has been around there for a long time is exactly what you need:
![handict_web_file111](https://github.com/goldendict/goldendict/assets/46510529/55088f50-ef4c-453e-a44a-3b10b1b151f3)
Doing so does not need to make a big fuss to satisfy someones' desire. And of course, it is also possible to do these jobs in a Program dictionary by caching search history.

**Comment by vedgy** [comment]
> From [xiaoyifang/goldendict-ng#417](https://github.com/xiaoyifang/goldendict-ng/pull/417)
> 
> I can copy that pull request here, but I don't know Qt enough to tell if it's good.

The file *globalbroadcaster.h* modified by that PR has been introduced in *goldendict-ng* and is not present here. So it's not a simple matter of copying.

**Comment by nonwill** [comment]
> > Compound expressions
> 
> This feature won't work with Japanese because sentences there have no spaces.

Yes, for Japanese, and also Chinese...  something like built-in **word segmentation** function is very necessary and a more general implementation can be done on the basis of **Compound expressions**.
